### ISSUE TITLE
Display glitch when switching the cursor to block

### ISSUE THREAD

**Issue by ehjmx** [raphamorim/rio#95]
At the request of @raphamorim I create this new issue. 

I'm on macOS 13.4 and I'm using the canary build

canary-4935590abfe47dab24d53dc1e2268a790b4b94fb

 There is display glitch when I switch to block for the cursor.

Cursor underscore looks like this
![cursor_as_underscore](https://github.com/raphamorim/rio/assets/75489369/a6635424-cc17-4a0b-9df5-6b2d7316567c)

Cursor as block the block uses two lines
![cursor_as_block](https://github.com/raphamorim/rio/assets/75489369/7928e24b-1717-4879-8e8b-3452c8a91fa0

Version 0.0.61 worked fine. Ffter updating to the canary version and making the necessary modifications to the config file even 0.0.61 did show this glitch after reverting back and using the old config.toml. 

_Originally posted by @ehjmx in https://github.com/raphamorim/rio/issues/92#issuecomment-1591613949_

**Comment by raphamorim** [comment]
@ehjmx can you please check if is still happening in latest main?

**Comment by ehjmx** [comment]
With the canary-a891a800901db9431e68c32c8f03a300aea06e0e build the cursor glitches are gone and so is the sometimes inconsistent use of the terminal height and width. 

For example ls used the full height and width but editors like joe, cmd line mail client aerc and artuin shell history only used the initial space that the terminal had when it was launched. 

I was about to create an issue on this once work is done for today.

**Comment by raphamorim** [comment]
That's great @ehjmx that fixed in the a891a800901db9431e68c32c8f03a300aea06e0e.

I haven't finished this fix yet. I am writing tests for the text styles in the renderer, so will avoid breaking this occasionally. 

Regarding the terminal height and width is that still happening? Would you mind exemplify a bit more?

**Comment by ehjmx** [comment]
The issue with the terminal heigh and width is still occassionally occuring, especially when using the 34 inch Monitor with his native 5k resolution.

This is the size of Rio when I open it
<img width="712" alt="Bildschirmfoto 2023-06-18 um 10 46 14" src="https://github.com/raphamorim/rio/assets/75489369/37640b16-9b9a-47ef-81c7-0acc7489a919">

Now I change the size 
<img width="1075" alt="Bildschirmfoto 2023-06-18 um 10 47 08" src="https://github.com/raphamorim/rio/assets/75489369/3c10aabd-b5d9-4a12-a995-844be3b11b62">

When I now open e.g. joe editor, atuin or aerc only the above initial size is used in the bigger window. 
When I use the vim, that comes with macOS or micro editor they use the full bigger window but at the cost of not using the theme that I set for Rio. Could this be related to issue #98 and/or because I set
`env-vars = ['SHELL=/opt/local/bin/bash']`
in my config file to use bash with Rio while I still have zsh as my default shell? 

Using canary canary-2316d6de8782f6eaa94121c508c7798b303ca415 does help on the build in retina display on the MacBookPro M2 2022, macOS 13.4. Tomorrow, Monday, I can check if the whole heigth and width are also used on my 5k Monitor.

**Comment by raphamorim** [comment]
Thank you @ehjmx for the detailed issue, this problem was happening before right? (Asking because I have been doing a major refactor of the renderer to port to WASM)

**Comment by ehjmx** [comment]
Yes @raphamorim the issue was before the canary build I mentioned. The refactoring did fix it. However I can only confirm this for the build in Retina display. For the 5k monitor at work I will check tomorrow.

**Comment by raphamorim** [comment]
Tested in a 5k monitor using v0.0.8 and it does look ok, closing this issue but let me know anything.
